pytest_flask_sqlalchemy/fixtures.py

```diff
@@ -70,7 +70,10 @@ def restart_savepoint(session, trans):
     @sa.event.listens_for(session, 'persistent_to_detached')
     @sa.event.listens_for(session, 'deleted_to_detached')
     def rehydrate_object(session, obj):
-        session.add(obj)
+        try:
+            session.add(obj)
+        except sa.exc.InvalidRequestError:
+            sa.orm.session.make_transient(obj)
```

Comment:
This seems right to me but I'm totally going to forget why we did this in the future, let's add a quick comment for posterity:

```suggestion
        except sa.exc.InvalidRequestError:
            # If the object has been deleted we need to add it back to the session context
            # See: https://github.com/sqlalchemy/sqlalchemy/blob/da4e4cb1c67230abd67e487cfb1bda8cd6910029/lib/sqlalchemy/orm/session.py#L2343
            sa.orm.session.make_transient(obj)
```